spec/support/test_app.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+require 'sinatra'
+require 'rack/auth/basic'
+
+helpers do
+  def protected!
+    return if authorized?
+
+    response['WWW-Authenticate'] = %(Basic realm="Restricted Area")
+    throw(:halt, [401, "Unauthorized access\nYou are denied access to this resource."])
+  end
+
+  def authorized?
+    @auth ||= Rack::Auth::Basic::Request.new(request.env)
+    @auth.provided? && @auth.basic? && @auth.credentials && @auth.credentials == %w[guest guest]
+  end
+end
+
+get '/' do
```

Comment:
nice one 👍 Fixing this has been on my list for months.. this looks like a nice long term solution